### PR TITLE
docs: Add API and CLI documentation

### LINKED ISSUE #3741: Engine, Playground and CLI Documentation Section
### What is the issue?

We need to have a section in the docs to explains the Dagger engine in more detail

### PR DESCRIPTION
**Pull request by vikram-dagger**
This commit adds documentation for the API and the CLI. This commit includes reference documentation for both, installation
steps for the CLI, tutorials and guides for both, and an introduction to the API Playground. This commit also updates the general FAQ with entries related to the GraphQL API.
    
This commit also adds a new ExternalButton component that produces a button linking to an external URL and opening that URL in a separate tab/window.

Closes #3741 , #3622,  #4024

Signed-off-by: Vikram Vaswani <vikram@dagger.io>

### CODE REVIEW

File: docs/current/api/990050-index.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+slug: /api
+---
+
+# Dagger API
```
Comment (vikram-dagger) [inline docs/current/api/990050-index.md:5]: I want to understand this terminology better in our specific context.
- Should this be "Dagger Engine" instead?
- What's the difference between the API and the Engine?
- Is the CLI an interface to the API, to the Engine, or both?

**Comment by vikram-dagger** [comment]
the linter is failing with

```
 #19 1.602 docs/current/cli/979595-reference.md:88 MD024/no-duplicate-heading/no-duplicate-header Multiple headings with the same content [Context: "#### Example"]
#19 1.602 docs/current/cli/979595-reference.md:100 MD024/no-duplicate-heading/no-duplicate-header Multiple headings with the same content [Context: "#### Usage"]
#19 1.602 docs/current/cli/979595-reference.md:106 MD024/no-duplicate-heading/no-duplicate-header Multiple headings with the same content [Context: "#### Options"]
```

I'm not sure we can avoid these duplicates for the reference page. Should we disable this check for this page, or for this PR?

**Comment by mircubed** [comment]
Feedback from 11/22 meeting with @shykes @kpenfound @jpadams @vikram-dagger @marcosnils  

### Overall feedback: 
- Single GraphQL API section - Make sure it says "GraphQL API" vs just API 
- Single CLI section with guides and refernce (not getting started guide) 
- Overview - 3 types of users  1) program with shell scripts 2) wrote own clients 3) use SDK, but want to go deeper 
- Get started guide that walks through typical things that you'd do in an SDK, but in graphQL. This should cover what you need to know for GraphQL if it is new to you (point to the existing GraphQL docs right now to save time) and then you can skip that section if you already know GraphQL. 
- Maybe point to the "getting started" GraphQL docs for now to save time and then we can write our own later. 
- Maybe use tab for the getting started guide - one for playground and one for CLI (for CLI, explain what it is and who it is for) 
- We need to update the docs.dagger.io overview page with a CTA to the API docs like we have for the SDKs. 
- mention two limitations for playground - playground is on our servers, so you shouldn't load any creds/secrets, we don't have access to your host.filesystem, and can't connect to local computer network 

### Command guides needed in the CLI section: 
- A single guide as "write Dagger using a shell script" with sub-guides below. The commands below would be their own guides. 
-  `Dagger query` examples needed  - this is the one we need for launch for sure. 
- `Dagger listen` - show how to `curl ` - this is one that we need for launch
- `Dagger exec` needs to be documented - this can wait until after launch, so we add this to the backlog 

Here are some shell examples to pull from: 
https://github.com/jpadams/dagger-jenkins-java/blob/main/Jenkinsfile
https://github.com/jpadams/dagger-jenkins-java/blob/main/dagger1gradle.sh
https://github.com/jpadams/dagger-jenkins-java/blob/main/queries.graphql


Next steps: 
@vikram-dagger get started on updates from notes above and work with @marcosnils on examples 
Next sync with @shykes will be Monday

File: docs/current/api/975146-concepts.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
+---
+slug: /api/concepts
+---
+
+# Key Concepts
```
Comment (vikram-dagger) [inline docs/current/api/975146-concepts.md:5]: I struggled a bit with how best to structure this tutorial. I decided finally on this concepts-based approach, as it seems to fit better for an API than the step-wise sequential walkthrough we have for the SDKs. I'd like to have feedback on whether this approach is OK or if I should consider a different structure altogether.

File: docs/current/api/975146-concepts.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+      }
+    }
+  }
+  ```
+
+  Here, the query requests the `stdout` field of the `Container` object returned in the previous step. To resolve this, Dagger will return a `File` object type containing the result of the last executed command. The output of the command is retrieved by addressing the `contents` field of the `File` type, which returns a `String`.
+
+## Lazy-Loading
+
+Dagger executes queries "lazily". This means that the individual build operations are performed only when a leaf value (scalar or enum) is requested. In the previous example, Dagger executes the query only when the `stdout` field is requested. This is a key advantage of Dagger: it allows users to write queries (or code, if using an SDK) in a procedural manner, then executes those queries in parallel for maximum speed.
```
Comment (vikram-dagger) [inline docs/current/api/975146-concepts.md:114]: I read through https://github.com/dagger/dagger/issues/3555, and I understand this might change. I'm assuming it's correct for now, but let me know if not. Also, is there a way for a user to identify in the API reference which fields trigger execution? If not, can you please provide a short list of common ones here?

File: docs/current/api/975146-concepts.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+          }
+        }
+      }
+    }
+  }
+  ```
+
+  Here, the query requests the `stdout` field of the `Container` object returned in the previous step. To resolve this, Dagger will return a `File` object type containing the result of the last executed command. The output of the command is retrieved by addressing the `contents` field of the `File` type, which returns a `String`.
+
+## Lazy-Loading
```
Comment (vikram-dagger) [inline docs/current/api/975146-concepts.md:112]: I would like feedback on which concepts should be discussed in this guide. The ones I've planned are lazy-loading and chaining, but maybe there are other (better?) ones?

File: docs/current/cli/752479-guides.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+---
+slug: /cli/guides
+---
+
+# Guides
+
+- [Proxy Requests to the Dagger API](./guides/632021-proxy-requests-api.md)
```
Comment (vikram-dagger) [inline docs/current/cli/752479-guides.md:7]: I know we discussed having both these guides at the root of the category, but I was wondering if we should move this into the Guides section. This is just guesswork but I think that "using a shell script" will be a much more common use case for the CLI than "proxying requests". WDYT?

File: docs/current/api/975146-concepts.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+# Key Concepts
+
+## Introduction
+
+One of Dagger's most powerful features is cross-language instrumentation: teams can use each other's tools without learning each other's language. Dagger achieves this by exposing its functionality through GraphQL, which serves as a low-level language-agnostic API framework for all Dagger operations.
+
+This approach has two main benefits:
+
+- It enables CI/CD pipeline authors to use whichever language feels natural and intuitive to them, without having to think about interoperability.
+- Rather than reimplementing the same automation in every possible programming language, it only needs to be implemented once and is then reusable by all. As a result, two cooperating teams no longer necessarily have to come to consensus on a common language.
```
Comment (shykes) [inline docs/current/api/975146-concepts.md:14]: This is very well-put, but sadly it will not be true until we ship extensions.

However, it does provide *some* interoperability benefits.

* Common runtime environment. Even if you write pipelines in different languages, they are all run by the same engine. So logging, scaling, monitoring, etc. only need to be solved once for everyone.
* Common low-level language. Even if teams can't directly re-use each other's pipelines (yet) they can take advantage of the fact that they all produce the same GraphQL queries. Similar to how developing for the same platform, even in different languages, creates opportunities for knowledge sharing and collaboration.

File: docs/current/api/975146-concepts.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+Let's start with a basic query:
+
+```graphql
+query {
+  container {
+    from(address: "alpine:latest") {
+      exec(args: ["apk", "info"]) {
+        stdout {
+          contents
+        }
```
Comment (shykes) [inline docs/current/api/975146-concepts.md:33]:
```suggestion
        stdout
```

As of today's release the API for `stdout` has changed.

File: docs/current/api/975146-concepts.md [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+1. Finally, consider the innermost level of the query:
+
+  ```graphql
+  query {
+    container {
+      from(address: "alpine:latest") {
+        exec(args: ["apk", "info"]) {
+          stdout {
+            contents
+          }
```
Comment (shykes) [inline docs/current/api/975146-concepts.md:103]:
```suggestion
          stdout
```

File: docs/current/api/975146-concepts.md (hunk 3 above)
Comment (shykes) [reply docs/current/api/975146-concepts.md:114]: There isn't a list of fields that trigger execution. But it would be useful to define a simple rule that people can memorize.

The rule I've been using so far is something like:

*The Dagger API is data-oriented. You query for data, and dagger returns that data. If executing a command is necessary to return that data, it will do so. When crafting a pipeline, ask yourself: what data do you need that pipeline to return? Is it an artifact? The result or standard output of a command? Most of the time, actually executing a command is just an implementation detail of what you actually want*

File: docs/current/api/990050-index.md [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+---
+
+# Dagger GraphQL API
+
+## What is the Dagger GraphQL API?
+
+Dagger exposes its functionality through a GraphQL API. GraphQL serves as a low-level language-agnostic API framework for all Dagger operations.
+
+In practice, interacting with the API directly is optional. You can instead use a Dagger SDK for one of the supported languages, or use the Dagger CLI.
+
```
Comment (shykes) [inline docs/current/api/990050-index.md:12]: A simple stack diagram here would be useful.

File: docs/current/cli/698277-index.md [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+The Dagger CLI may be a good fit if you are...
+
+* A "designated devops person" who doesn't think of themselves as a developer - nothing makes you happier than a battle-hardened shell script.
+* A GraphQL veteran looking to connect existing GraphQL tools and libraries with Dagger's API.
+
+The Dagger CLI may *not* be a good fit if you are...
+
+* A developer who doesn't know shell scripting, and is not interested in learning it.
+* Someone who loves writing YAML all day, thank you very much.
+* A container skeptic: the less containers are involved, the happier you are.
```
Comment (shykes) [inline docs/current/cli/698277-index.md:35]: I think we should cut this part. The CLI is different from the other sections, as every Dagger user may benefit from using, although not always for the same reasons. In that way it is different from both the SDKs and GraphQL API.

For example:

* Non-developers can use it to interact with the Dagger engine from the terminal or a shell script
* Developers using an SDK can use it as a complement to their development experience (trying out queries, debugging their code, etc)
* Developers *not* using an SDK can use it in the same way, but will need it even more given their reliance on the GraphQL API.

File: docs/current/cli/752479-guides.md (hunk 5 above)
Comment (shykes) [inline docs/current/cli/752479-guides.md:7]: Heads up, we might cut `dagger listen` entirely for security reasons. Discussion underway https://github.com/dagger/dagger/issues/3830#issuecomment-1325629630.

At the very least I would de-prioritize writing this guide versus others.

File: docs/current/cli/979595-reference.md [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+```shell
+dagger listen
+```
+
+Start Dagger Engine on port 1234:
+
+```shell
+dagger listen --listen 1234
+```
+
```
Comment (shykes) [inline docs/current/cli/979595-reference.md:125]: Same heads up as above: we may cut `dagger listen`

File: docs/current/cli/guides/632021-proxy-requests-api.md [hunk 12]
```diff
@@ -0,0 +1,6 @@
+---
+slug: /cli/632021/proxy-requests-api
+displayed_sidebar: 'current'
+---
+
+# Proxy Requests to the Dagger API
```
Comment (shykes) [inline docs/current/cli/guides/632021-proxy-requests-api.md:6]: We may cut `dagger listen`. This guide is still useful, but will rely on other commands.

**Review by shykes — APPROVED** [review]
Looking good. I left some comments and edits.

File: docs/current/api/975146-concepts.md (hunk 7 above)
Comment (vikram-dagger) [reply docs/current/api/975146-concepts.md:33]: Fixed

File: docs/current/api/975146-concepts.md (hunk 6 above)
Comment (vikram-dagger) [reply docs/current/api/975146-concepts.md:14]: Added

File: docs/current/api/975146-concepts.md (hunk 8 above)
Comment (vikram-dagger) [reply docs/current/api/975146-concepts.md:103]: Fixed

File: docs/current/cli/698277-index.md (hunk 10 above)
Comment (vikram-dagger) [reply docs/current/cli/698277-index.md:35]: Deleted

File: docs/current/cli/752479-guides.md (hunk 5 above)
Comment (vikram-dagger) [reply docs/current/cli/752479-guides.md:7]: Understood. I'll leave the stubs and navigation in for now, we can finalize this based on the discussion result.

File: docs/current/cli/979595-reference.md (hunk 11 above)
Comment (vikram-dagger) [reply docs/current/cli/979595-reference.md:125]: OK

File: docs/current/cli/guides/632021-proxy-requests-api.md (hunk 12 above)
Comment (vikram-dagger) [reply docs/current/cli/guides/632021-proxy-requests-api.md:6]: OK

File: docs/current/api/975146-concepts.md [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+      }
+    }
+  }
+  ```
+
+  Here, the query requests the `stdout` field of the `Container` object returned in the previous step. To resolve this, Dagger will return a `File` object type containing the result of the last executed command. The output of the command is retrieved by addressing the `contents` field of the `File` type, which returns a `String`.
+
+## Lazy-Loading
+
+Dagger executes queries "lazily". This means that the individual build operations are performed only when a leaf value (scalar or enum) is requested. In the previous example, Dagger executes the query only when the `stdout` field is requested. This is a key advantage of Dagger: it allows users to write queries (or code, if using an SDK) in a procedural manner, then executes those queries in parallel for maximum speed.
```
[the remark this answers is not in the document]
Comment (vikram-dagger) [reply docs/current/api/975146-concepts.md]: @sipsma added more detail in this section. Please review and add feedback.

File: docs/current/cli/465058-install.md [hunk 14]
```diff
@@ -0,0 +1,129 @@
+---
+slug: /cli/465058/install
```
Comment (vikram-dagger) [inline docs/current/cli/465058-install.md:2]: @marcosnils can you please review the installation steps and confirm they are fine? I have assumed the same as the previous CLI.

File: docs/current/162770-faq.md [hunk 15]
```diff
@@ -12,6 +12,18 @@ We currently offer technical previews of a [Go SDK](/sdk/go) and a [Python SDK](
 
 Dagger SDKs use your existing Docker credentials and do not require separate authentication. Simply execute `docker login` against your container registry on the host where your Dagger pipelines are running.
 
+### What API framework does Dagger use?
```
Comment (marcosnils) [inline docs/current/162770-faq.md:15]:
```suggestion
### What API query language does Dagger use?
```

File: docs/current/162770-faq.md [hunk 16]
```diff
@@ -12,6 +12,18 @@ We currently offer technical previews of a [Go SDK](/sdk/go) and a [Python SDK](
...
+
+Dagger uses GraphQL as its low-level language-agnostic API framework.
```
Comment (marcosnils) [inline docs/current/162770-faq.md:17]:
```suggestion
Dagger uses GraphQL as its low-level language-agnostic API query language.
```

File: docs/current/162770-faq.md [hunk 17]
```diff
@@ -12,6 +12,18 @@ We currently offer technical previews of a [Go SDK](/sdk/go) and a [Python SDK](
...
+
+### Do I need to know GraphQL for Dagger?
```
Comment (marcosnils) [inline docs/current/162770-faq.md:19]:
```suggestion
### Do I need to know GraphQL to use Dagger?
```

File: docs/current/162770-faq.md [hunk 18]
```diff
@@ -12,6 +12,18 @@ We currently offer technical previews of a [Go SDK](/sdk/go) and a [Python SDK](
...
+
+No. You only need to know one of Dagger's supported languages to use Dagger. Communicating with the Dagger API is as simple as calling Dagger SDK functions in the language you choose to use. The conversion of those function calls to language-agnostic API calls is performed by the Dagger SDK. You do not need to be an expert in GraphQL to use Dagger.
```
Comment (marcosnils) [inline docs/current/162770-faq.md:21]:
```suggestion
No. You only need to know one of Dagger's supported SDKs languages to use Dagger. The translation to underlying GraphQL  API calls  is handled internally by the Dagger SDK of your choice. 
```

File: docs/current/162770-faq.md [hunk 19]
```diff
@@ -12,6 +12,18 @@ We currently offer technical previews of a [Go SDK](/sdk/go) and a [Python SDK](
...
+
+### There's no SDK for &lt;language&gt; yet. Can I still use Dagger?
+
+Yes. It's possible to use the Dagger API from any language that [supports GraphQL](https://github.com/chentsulin/awesome-graphql) or from the Dagger CLI.
```
Comment (marcosnils) [inline docs/current/162770-faq.md:25]:
```suggestion
Yes. It's possible to use the Dagger API from any language that [supports GraphQL](https://graphql.org/code/) or from the Dagger CLI.
```

Add a link to the `Dagger CLI` docs?

File: docs/current/api/282105-playground.md [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+slug: /api/playground
+---
+
+import Button from "@site/src/components/atoms/button.js";
+import LinkCTA from "@site/src/components/atoms/linkCTA.js";
+import ButtonExternal from "@site/src/components/atoms/buttonExternal.js";
+
+# Playground
+
+The API Playground is an in-browser tool for writing, validating, and testing Dagger GraphQL API queries.
```
Comment (marcosnils) [inline docs/current/api/282105-playground.md:11]:
```suggestion
The API Playground is an in-browser tool for testing, running and sharing Dagger GraphQL API queries.
```

File: docs/current/api/975146-concepts.md [hunk 21]
```diff
@@ -0,0 +1,126 @@
+---
+slug: /api/concepts
+---
+
+# Key Concepts
+
+## Introduction
+
+One of Dagger's most powerful features is cross-language instrumentation: teams can use each other's tools without learning each other's language. Dagger achieves this by exposing its functionality through GraphQL, which serves as a low-level language-agnostic API framework for all Dagger operations.
```
Comment (marcosnils) [inline docs/current/api/975146-concepts.md:9]:
```suggestion
One of Dagger's key aspects is cross-language instrumentation: teams can leverage their familiar languages  without the need to learn a new SDL to write their CI/CD pipelines . Dagger allows this by exposing its functionality through GraphQL which serves as a low-level language-agnostic API framework for all Dagger operations.
```

File: docs/current/api/975146-concepts.md [hunk 22]
```diff
@@ -0,0 +1,126 @@
...
+
+## Objects, Fields and Build Operations
+
+Let's start with a basic query:
+
+```graphql
+query {
+  container {
+    from(address: "alpine:latest") {
+      exec(args: ["apk", "info"]) {
```
Comment (marcosnils) [inline docs/current/api/975146-concepts.md:31]:
```suggestion
      withExec(args: ["apk", "info"]) {
```

File: docs/current/api/975146-concepts.md [hunk 23]
```diff
@@ -0,0 +1,126 @@
...
+This query represents a very simple Dagger pipeline. In plain English, it instructs Dagger to "download the latest `alpine` container image, run the command `apk info` in that image, and print the results of the command to the standard output device".
+
+Here's an example of the command output - a list of the packages installed in the image:
+
+```graphql
+{
+ "data": {
+  "container": {
+   "from": {
+    "exec": {
```
Comment (marcosnils) [inline docs/current/api/975146-concepts.md:48]:
```suggestion
    "withExec": {
```

File: docs/current/api/975146-concepts.md [hunk 24]
```diff
@@ -0,0 +1,126 @@
...
+     }
+    }
+   }
+  }
+ }
+}
+```
+
+A GraphQL schema works by defining one or more object types, and then fields (and field arguments) on those object types. The fields of an object type can themselves be objects, allowing for different entities to be logically connected with each other. API users can then perform GraphQL queries to return all or some fields of an object.
+
```
[the remark this answers is not in the document]
Comment (marcosnils) [reply docs/current/api/975146-concepts.md:60]: > * It's almost comparable to a bash pipeline where you pass data along and each step manipulates it in some way

by bash pipeline you mean piping multiple commands? I guess that analogy is correct, since basically it's a pipe of transformations to the initial state. Isn't "reactive programming" (https://en.wikipedia.org/wiki/Reactive_programming) the closest term?

File: docs/current/api/975146-concepts.md [hunk 25]
```diff
@@ -0,0 +1,126 @@
...
+  This query requests the `from` field of Dagger's `Container` object type, passing it the address of a container image as argument. To resolve this, Dagger will initialize a container from the image published at the given address and return a `Container` object representing the new container image.
+
+1. Now, consider the next level of the query:
+
+  ```graphql
+  query {
+    container {
+      from(address: "alpine:latest") {
+        exec(args: ["apk", "info"]) {
```
Comment (marcosnils) [inline docs/current/api/975146-concepts.md:83]:
```suggestion
        withExec(args: ["apk", "info"]) {
```

File: docs/current/api/975146-concepts.md [hunk 26]
```diff
@@ -0,0 +1,126 @@
...
+
+  Here, the query requests the `exec` field of the `Container` object returned in the previous step, passing it the command to be executed as an array of arguments. To resolve this, Dagger will  define the command for execution in the container image and return a `Container` object containing the execution results.
+
+1. Finally, consider the innermost level of the query:
+
+  ```graphql
+  query {
+    container {
+      from(address: "alpine:latest") {
+        exec(args: ["apk", "info"]) {
```
Comment (marcosnils) [inline docs/current/api/975146-concepts.md:99]:
```suggestion
        withExec(args: ["apk", "info"]) {
```

File: docs/current/api/975146-concepts.md [hunk 27]
```diff
@@ -0,0 +1,126 @@
...
+  }
+  ```
+
+  Here, the query requests the `stdout` field of the `Container` object returned in the previous step. To resolve this, Dagger will return a `File` object type containing the result of the last executed command. The output of the command is retrieved by addressing the `contents` field of the `File` type, which returns a `String`.
+
+## Lazy-Loading
+
+Dagger executes queries "lazily". This means that the individual build operations are performed only when a leaf value (scalar or enum) is requested. In the previous example, Dagger executes the query only when the `stdout` field is requested. This is a key advantage of Dagger: it allows users to write queries (or code, if using an SDK) in a procedural manner, then executes those queries in parallel for maximum speed.
+
+[example TBD]
```
[the remark this answers is not in the document]
Comment (marcosnils) [reply docs/current/api/975146-concepts.md:118]: We've synced with Vikram today and we have a really nice example to showcase this :hugs:

File: docs/current/api/990050-index.md [hunk 28]
```diff
@@ -0,0 +1,30 @@
...
+slug: /api
+---
+
+# Dagger GraphQL API
+
+## What is the Dagger GraphQL API?
+
+Dagger exposes its functionality through a GraphQL API. GraphQL serves as a low-level language-agnostic API framework for all Dagger operations.
+
+In practice, interacting with the API directly is optional. You can instead use a Dagger SDK for one of the supported languages, or use the Dagger CLI.
```
Comment (marcosnils) [inline docs/current/api/990050-index.md:11]:
```suggestion
In practice, interacting with the API directly is optional. We encourage users to use instead a Dagger SDK for one of the supported languages, or use the Dagger CLI.
```

File: docs/current/api/990050-index.md (hunk 9 above)
Comment (marcosnils) [reply docs/current/api/990050-index.md:12]: https://github.com/mermaid-js/mermaid/issues/3307  :eyes:

File: docs/current/cli/465058-install.md (hunk 14 above)
Comment (marcosnils) [reply docs/current/cli/465058-install.md:2]: Yes, they are the same. I'll double check just to be sure

File: docs/current/cli/465058-install.md [hunk 29]
```diff
@@ -0,0 +1,129 @@
...
+{label: 'macOS', value: 'macos'}, {label: 'Linux', value: 'linux'}, {label: 'Windows', value: 'windows'},
+]}>
+
+<TabItem value="macos">
+
+We assume that you have [Homebrew](https://brew.sh/) installed.
+If you do, you can install `dagger` with a single command:
+
+```shell
+brew install dagger
```
Comment (marcosnils) [inline docs/current/cli/465058-install.md:36]: We should encourage users to use our tap instead of installing from the bottle formuale. Check instructions here https://docs.dagger.io/1001/install/

File: docs/current/cli/465058-install.md [hunk 30]
```diff
@@ -0,0 +1,129 @@
...
+# macOS ARM:
+dagger is /opt/homebrew/bin/dagger
+# macOS Intel:
+dagger is /usr/local/bin/dagger
+```
+
+If you do not have Homebrew installed, or you want to install a specific version of `dagger`, you can run:
+
+```shell
+curl -L https://dl.dagger.io/dagger/install.sh | VERSION=0.2.232 sh
```
Comment (marcosnils) [inline docs/current/cli/465058-install.md:52]: I don't think we have bumped the install script to support the new dagger CLI.  @gerhard can you help me verify this? I'm not familiarized with this flow :raised_hands:

File: docs/current/cli/465058-install.md [hunk 31]
```diff
@@ -0,0 +1,129 @@
...
+
+./bin/dagger version
+dagger 0.2.232 (GIT_SHA) darwin/arm64
```
Comment (marcosnils) [inline docs/current/cli/465058-install.md:55]: This version will be different, we should update it once we release the new CLI.

File: docs/current/cli/465058-install.md [hunk 32]
```diff
@@ -0,0 +1,129 @@
...
+
+</TabItem>
+
+<TabItem value="linux">
+
+The quickest way of installing `dagger` on Linux is to run the following command:
+
+```shell
+cd /usr/local
+curl -L https://dl.dagger.io/dagger/install.sh | sh
```
Comment (marcosnils) [inline docs/current/cli/465058-install.md:66]: Same as above regarding install script. cc @gerhard

**Comment by marcosnils** [comment]
@vikram-dagger did a first pass and left some comments. LMK if you need to sync.

File: docs/current/api/975146-concepts.md (hunk 27 above)
[the remark this answers is not in the document]
Comment (vikram-dagger) [reply docs/current/api/975146-concepts.md]: Added this now

File: docs/current/cli/752479-guides.md (hunk 5 above)
Comment (vikram-dagger) [reply docs/current/cli/752479-guides.md:7]: Nix, I'm removing the stubs and links to this guide to make this PR ready to merge. I will retain `dagger listen` in the CLI reference for the moment.

File: docs/current/api/975146-concepts.md (hunk 24 above)
[the remark this answers is not in the document]
Comment (vikram-dagger) [reply docs/current/api/975146-concepts.md:60]: I've incorporated some of @sipsma 's comments in the draft now. Thanks.

File: website/sidebars.js [hunk 33]
```diff
@@ -98,6 +98,46 @@ module.exports = {
...
+        {
+          type: "doc",
+          label: "Overview",
+          id: "current/api/index",
+        },
+        "current/api/concepts",
+        "current/api/playground",
+        {
+          type: "link",
+          label: "Reference",
```
Comment (vikram-dagger) [inline website/sidebars.js:116]: @crjm could you please update #3745 to ensure the static API reference doc is available at this URL? Thanks

Comment (crjm) [reply website/sidebars.js:116]: > @crjm could you please update #3745 to ensure the static API reference doc is available at this URL? Thanks

Sure!

File: docs/current/cli/465058-install.md (hunk 29 above)
Comment (vikram-dagger) [reply docs/current/cli/465058-install.md]: Done

**Comment by kpenfound** [comment]
Should the "Try It!" button instead be the same as "Try it in the API Playground!" button in #3745 ?

File: docs/current/cli/752479-guides.md (hunk 5 above)
Comment (vikram-dagger) [reply docs/current/cli/752479-guides.md]: Fixed and removed dagger listen

File: docs/current/api/975146-concepts.md [hunk 34]
```diff
@@ -0,0 +1,310 @@
...
+    "container": {
+      "withExec": {
+        "stdout": "myfile\n"
+      }
+    }
+  }
+}
+```
+
+As this example demonstrates, Dagger object IDs hold the state of their corresponding object. This state can be transferred from one query to another, or even from one Dagger instance to another. It can also be updated or used as input to other objects.
```
[the remark this answers is not in the document]
Comment (vikram-dagger) [reply docs/current/api/975146-concepts.md:223]: Fixed

File: docs/current/api/975146-concepts.md [hunk 35]
```diff
@@ -0,0 +1,310 @@
...
+  }
+  ```
+
+  Here, the query requests the `stdout` field of the `Container` object returned in the previous step. To resolve this, Dagger will return a `String` containing the result of the last executed command.
+
+## Object Identifiers
+
+In a GraphQL schema, every object exposes an `id` field. This ID serves to uniquely identify and retrieve an object, and is also used by GraphQL's caching mechanism.
+
+In the Dagger GraphQL API too, objects expose an ID but here, the ID is actually a base64-encoded block representing the object's state at a given time. Objects like `Container` and `Directory` should be thought of as collections of state, which are updated by subsequent field resolutions and whose ID represents their state at the instant of field resolution.
```
[the remark this answers is not in the document]
Comment (vikram-dagger) [reply docs/current/api/975146-concepts.md:117]: Fixed

File: docs/current/api/975146-concepts.md [hunk 36]
```diff
@@ -0,0 +1,310 @@
...
+
+This example demonstrates the lazy evaluation model in action. The client requests data, and the Dagger GraphQL API returns that data. If executing a command is necessary to return that data, it will do so; if not, it will not.
+
+:::note
+Lazy evaluation is a key advantage of GraphQL. It allows users to write queries (or code, if using an SDK) in a procedural manner, and resolves those queries only when necessary to return data to the user. Dagger leverages this lazy evaluation model to optimize and parallelize query execution for maximum speed and performance.
+:::
+
+## Conclusion
+
+This tutorial introduced you to the Dagger GraphQL API. It explained the reasons for using GraphQL and walked you through the components of a Dagger GraphQL API query. It also explained how Dagger leverages some of GraphQL's features, such as unique identifiers and lazy evaluation, to improve the portability and speed of CI/CD pipeline execution..
```
[the remark this answers is not in the document]
Comment (vikram-dagger) [reply docs/current/api/975146-concepts.md:308]: Fixed

File: docs/current/api/975146-concepts.md [hunk 37]
```diff
@@ -0,0 +1,310 @@
...
+    }
+  }
+}
+```
+
+As this example demonstrates, Dagger object IDs hold the state of their corresponding object. This state can be transferred from one query to another, or even from one Dagger instance to another. It can also be updated or used as input to other objects.
+
+## Lazy Evaluation
+
+GraphQL executes queries "lazily". This means that query resolution is triggered only when a leaf value (scalar) is requested. In the previous example, Dagger resolves the query and performs the build operations only when the `stdout` field is requested.
```
[the remark this answers is not in the document]
Comment (vikram-dagger) [reply docs/current/api/975146-concepts.md:227]: Modified
